core/controllers/acl_decorators.py

```diff
@@ -2413,12 +2413,28 @@ def test_can_access(self, topic_id, subtopic_id, **kwargs):
         Raises:
             PageNotFoundException: The given page cannot be found.
         """
+        topic = topic_fetchers.get_topic_by_name(topic_name)
+        if topic is None:
+            raise self.PageNotFoundException
+
+        topic_rights = topic_services.get_topic_rights(topic.id)
+        if topic_rights is None or not topic_rights.topic_is_published:
+            raise self.PageNotFoundException
+
+        subtopic_is_present = False
```

Comment:
subtopic_is_present = any(subtopic.id == int(subtopic_id) for subtopic in topic.subtopics)

Reply:
Done